.gitattributes

```diff
@@ -0,0 +1,9 @@
+# By default, do not attempt to do any end-of-line conversion upon
+# checkin or checkout.
+*       -text
+
+# For specific file-types, force conversion to LF line endings. This
+# can be overridden below or in a more specific .gitattributes if, for
+# example, we want to allow a .txt file to contain a CRLF line ending
+*.go    text eol=lf
+*.txt   text eol=lf
```

Comment:
A perhaps simpler option would be just treating all files as non-text, like what Go does. Then you rely on any developers on Windows to have their editors set up to use LF and not CRLF. But I see the appeal of your setup - you enforce Go and text files to never use CRLF.

Reply:
Per the default at the top of this file, I agree that's exactly what we want.

But per https://github.com/golang/go/blob/2b58ca6e3de3d93817a4e6cc55378015eb3c2040/.gitattributes#L5-L6 we would then require that we have some additional check to verify that for `.go` and `.txt` files (there could be others that need to fall into this category) don't accidentally end up with CRLF.

So I think this is what we (for now!)